### PR TITLE
Store

### PR DESCRIPTION
**Pull request by joseims**

### CODE REVIEW

File: main-character.cpp [hunk 1]
```diff
@@ -7,9 +7,10 @@ struct main_character {
     int strength;
     int defense;
     int health;
-    int maxHealth;
+    int max_health;
     item weapon;
     item armor;
-    int game_progress_multiplier;
+    int game_progress_multiplyer;
```
Comment (raonismaneoto) [inline main-character.cpp]: multiplier*

File: main-character.cpp [hunk 2]
```diff
@@ -7,9 +7,10 @@ struct main_character {
...
+    int coins;
     int potion;
-};
+} character;
```
Comment (raonismaneoto) [inline main-character.cpp]: Does it work?

File: store.cpp [hunk 3]
```diff
@@ -1,8 +1,168 @@
 #include <bits/stdc++.h>
 using namespace std;
```
Comment (raonismaneoto) [inline store.cpp]: put it after the imports

File: store.cpp [hunk 4]
```diff
@@ -1,8 +1,168 @@
...
+#include "item-generator.cpp"
+#include "main-character.cpp"
 
 item itens[3];
-void seeStore();
-void exitStore();
-void buyItem();
\ No newline at end of file
+bool bought[3];
+const int potion_price = 5;
```
Comment (raonismaneoto) [inline store.cpp]: we're declaring consts with CAPITAL letters

File: store.cpp [hunk 5]
```diff
@@ -1,8 +1,168 @@
...
+int actual_potion_price;
+const string potion_text = "Poção (Recupera X de vida)";
+const string store_welcome_text = "BEM VINDO MEU PARCEIRO\n";
+const string already_bought_text = "Você já comprou esse item, escolha novamente\n";
+const string exit_text = "Xau xau meu parceiro\n";
+const string not_enought_coins_text = "Sinto muito, você não me parece ter dinheiro suficiente para comprar isso\n";
```
Comment (raonismaneoto) [inline store.cpp]: enough*

File: store.cpp [hunk 6]
```diff
@@ -1,8 +1,168 @@
...
+const string invalid_option_text= "Opção Inválida, escolha novamente\n";
```
Comment (raonismaneoto) [inline store.cpp]: make sure you're using the same space before and after the =

File: store.cpp [hunk 7]
```diff
@@ -1,8 +1,168 @@
...
+void see_store(main_character character) {
+    calculate_actual_potion_price(character);
+    printf("%s",store_welcome_text );
+    int game_progress_multiplyer = character.game_progress_multiplyer;
+    reset_is_bought();
+    itens[0] = generate_weapon(game_progress_multiplyer);
+    itens[1] = generate_armor(game_progress_multiplyer);
+    itens[2] = generate_rand_item(game_progress_multiplyer);
+    run_store(character,game_progress_multiplyer);
+
```
Comment (raonismaneoto) [inline store.cpp]: ?

File: store.cpp [hunk 8]
```diff
@@ -1,8 +1,168 @@
...
+const string potion_text = "Poção (Recupera X de vida)";
+const string store_welcome_text = "BEM VINDO MEU PARCEIRO\n";
+const string already_bought_text = "Você já comprou esse item, escolha novamente\n";
+const string exit_text = "Xau xau meu parceiro\n";
+const string not_enought_coins_text = "Sinto muito, você não me parece ter dinheiro suficiente para comprar isso\n";
+const string invalid_option_text= "Opção Inválida, escolha novamente\n";
+
+void see_store(main_character character) {
+    calculate_actual_potion_price(character);
+    printf("%s",store_welcome_text );
```
Comment (raonismaneoto) [inline store.cpp]: you have to keep a pattern, there is no space between the variable and the comma here, but there is between the variable and )

File: store.cpp [hunk 9]
```diff
@@ -1,8 +1,168 @@
...
+    for (int i = 0; i< 3; i ++) {
+        bought[i] = false;
+    }
+};
+
+void exit_store() {
+    printf("%s",exit_text);
+};
+
+void buy_item(main_character character, item item,int index) {
```
Comment (raonismaneoto) [inline store.cpp]: missing a space after the last comma

File: store.cpp [hunk 10]
```diff
@@ -1,8 +1,168 @@
...
+        character.coins -= item.price;
+        equip_item(character,item,index);
+        bought[index] = true;
+    } else {
+        printf("%s",not_enought_coins_text);
+    }
+    run_store(character,character.game_progress_multiplyer);
+};
+
+void equip_item(main_character character, item item, int index) {
```
Comment (raonismaneoto) [inline store.cpp]: rename this item param to avoid any misunderstanding

File: store.cpp [hunk 11]
```diff
@@ -1,8 +1,168 @@
...
+    run_store();
+};
+
+void buying_menu(main_character character) {
+     int option;
+     while (true) {
+        scanf("%d",&option);
+        if (is_bought(option)) {
+            if (option == 1) {
+                buy_item(character,itens[0],0);
```
Comment (raonismaneoto) [inline store.cpp]: add spaces between the params

File: store.cpp [hunk 12]
```diff
@@ -1,8 +1,168 @@
...
+    }
+    run_store();
+};
+
+void buying_menu(main_character character) {
+     int option;
+     while (true) {
+        scanf("%d",&option);
+        if (is_bought(option)) {
+            if (option == 1) {
```
Comment (raonismaneoto) [inline store.cpp]: don't you think that a switch case would be better in this case? I guess it would be much more readable

File: store.cpp [hunk 13]
```diff
@@ -1,8 +1,168 @@
...
+                buy_potion(character);
+                return;
+            }  else if (option == 5) {
+                exit_store();
+                return;
+            }  else {
+                printf("%s",invalid_option_text);
+            }
+        } else {
+            printf("%s",already_bought_text);
```
Comment (raonismaneoto) [inline store.cpp]: space after the comma

**Review by raonismaneoto — CHANGES_REQUESTED** [review]

File: store.cpp (hunk 12 above)
Comment (joseims) [reply store.cpp]: nop, dont like then

File: store.cpp [hunk 14]
```diff
@@ -1,8 +1,181 @@
 #include <bits/stdc++.h>
 using namespace std;
-#include "weapon-generator.cpp"
+#include "main-character.cpp"
+
+
 
 item itens[3];
```
Comment (vsseixaso) [inline store.cpp]: exchange itens for items

File: store.cpp [hunk 15]
```diff
@@ -1,8 +1,174 @@
-#include <bits/stdc++.h>
 using namespace std;
-#include "weapon-generator.cpp"
+
+
 
 item itens[3];
```
Comment (vsseixaso) [inline store.cpp:5]: exchange itens for items

File: store.cpp [hunk 16]
```diff
@@ -1,8 +1,174 @@
...
+const string STORE_WELCOME_TEXT = "BEM VINDO MEU PARCEIRO\n";
+const string ALREADY_BOUGHT_TEXT = "Você já comprou esse item, escolha novamente\n";
+const string EXIT_TEXT = "Xau xau meu parceiro\n";
+const string NOT_ENOUGH_COINS_TEXT = "Sinto muito, você não me parece ter dinheiro suficiente para comprar isso\n";
+const string INVALID_OPTION_TEXT = "Opção Inválida, escolha novamente\n";
+
+
+void run_store(main_character main_character);
+
+void print_potion(int game_progress_multiplier) {
```
Comment (vsseixaso) [inline store.cpp]: exchange game_progress_multiplier for game_progress_multiplyer everywhere

**Review by raonismaneoto — APPROVED** [review]

**Pull request merged**